libs/ui/tabs/brain/src/lib/brn-tabs-list.directive.ts

```diff
@@ -25,14 +25,13 @@ export class BrnTabsListDirective implements AfterContentInit {
 
 	private _keyManager?: FocusKeyManager<BrnTabsTriggerDirective>;
 
-	@ContentChildren(BrnTabsTriggerDirective, { descendants: true })
-	public triggers?: QueryList<BrnTabsTriggerDirective>;
+	public triggers = contentChildren(BrnTabsTriggerDirective, { descendants: true });
 
 	public ngAfterContentInit() {
 		if (!this.triggers) {
```

Comment:
I let the type definition be implied from `contentChildren` and it looks like it is always a `Signal<ReadonlyArray<T>>` and it won't be `undefined` even when accessed in constructor. Then it will be a Signal with an empty array.

Compared to `@ContentChildren(BrnTabsTriggerDirective, { descendants: true })` this might be `undefined` when access in the constructor.

![CleanShot 2024-11-12 at 12 23 03](https://github.com/user-attachments/assets/5187fe72-443d-4a67-872c-b41dfe1c715d)

https://stackblitz.com/edit/stackblitz-starters-urvbjs?file=src%2Fmain.ts

Reply:
Not sure if I correctly understand your question. I don't think we need to add `?` to the variable to add additional checks if its available or not. Maybe I wrong, but while using it on spartan page it seems to work without any issues.